clang/lib/Driver/ToolChains/Arch/X86.h

```diff
@@ -21,13 +21,29 @@ namespace driver {
 namespace tools {
 namespace x86 {
 
+enum class DecimalFloatABI {
+  Default,    // Target-specific default.
+  None,       // No decimal floating-point support.
+  Libgcc_BID, // libgcc with BID encoding.
+  Libgcc_DPD, // libgcc with DPD encoding.
+  Hard,       // Target-specific hard ABI.
+  Invalid
```

Comment:
What is the motivation for an `Invalid` enumerator? A comment regarding its intended purpose would be helpful. My intuition is that an invalid ABI (whether an unrecognized argument provided with `-mdecimal-float-abi` or a unsupported ABI for a given target) would be diagnosed and the compiler invocation aborted.